reactivesocket-transport-tcp/src/main/java/io/reactivesocket/transport/tcp/server/ReactiveSocketServerHandler.java

```diff
@@ -28,51 +26,51 @@
 import io.reactivesocket.LeaseGovernor;
 import io.reactivesocket.ReactiveSocket;
 import io.reactivesocket.transport.tcp.MutableDirectByteBuf;
-import org.agrona.BitUtil;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.util.concurrent.ConcurrentHashMap;
+import static org.agrona.BitUtil.SIZE_OF_INT;
 
-@ChannelHandler.Sharable
 public class ReactiveSocketServerHandler extends ChannelInboundHandlerAdapter {
-    private Logger logger = LoggerFactory.getLogger(ReactiveSocketServerHandler.class);
-
-    private ConcurrentHashMap<ChannelId, ServerTcpDuplexConnection> duplexConnections = new ConcurrentHashMap<>();
+    private static final Logger logger = LoggerFactory.getLogger(ReactiveSocketServerHandler.class);
+    private static final int MAX_FRAME_LENGTH = Integer.MAX_VALUE >> 1;
 
     private ConnectionSetupHandler setupHandler;
-
     private LeaseGovernor leaseGovernor;
+    private ServerTcpDuplexConnection connection;
 
     protected ReactiveSocketServerHandler(ConnectionSetupHandler setupHandler, LeaseGovernor leaseGovernor) {
         this.setupHandler = setupHandler;
         this.leaseGovernor = leaseGovernor;
+        this.connection = null;
     }
 
     public static ReactiveSocketServerHandler create(ConnectionSetupHandler setupHandler) {
         return create(setupHandler, LeaseGovernor.UNLIMITED_LEASE_GOVERNOR);
     }
 
     public static ReactiveSocketServerHandler create(ConnectionSetupHandler setupHandler, LeaseGovernor leaseGovernor) {
-        return new
-            ReactiveSocketServerHandler(
-            setupHandler,
-            leaseGovernor);
-
+        return new ReactiveSocketServerHandler(setupHandler, leaseGovernor);
     }
 
     @Override
     public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
         ChannelPipeline cp = ctx.pipeline();
         if (cp.get(LengthFieldBasedFrameDecoder.class) == null) {
-            ctx
-                .pipeline()
-                .addBefore(
-                    ctx.name(),
-                    LengthFieldBasedFrameDecoder.class.getName(),
-                    new LengthFieldBasedFrameDecoder(Integer.MAX_VALUE >> 1, 0, BitUtil.SIZE_OF_INT, -1 * BitUtil.SIZE_OF_INT, 0));
+            LengthFieldBasedFrameDecoder frameDecoder =
```

Comment:
Minor: It doesn't make much of a difference but we can move this to server initialization i.e. add these handlers upfront using `ChannelInitializer` instead of doing it on `handlerAdded`